src/utils/universalProvider.ts

```diff
@@ -62,11 +63,9 @@ export class UniversalProviderFactory {
     )
 
     // Subscribe to session delete
-    UniversalProviderFactory.provider.on(
-      'session_delete',
-      ({ id, topic }: { id: number; topic: string }) => {
-        console.log(id, topic)
-      }
-    )
+    UniversalProviderFactory.provider.on('session_delete', () => {
+      delete UniversalProviderFactory.provider?.session.namespaces.solana
```

Comment:
Double verify, but I think sign sdk does this deletion under the hood?

Reply:
It's done in cases where a failure somewhere along the pipeline happens. Disconnected from the wallet but the dApp is not aware. Happened once or twice and don't wanna risk it